3rdparty/openssl/CMakeLists.txt

```diff
@@ -32,33 +39,43 @@ else ()
   message(FATAL_ERROR "Unknown architecture. OpenSSL cannot be configured.")
 endif ()
 
+set(OPENSSL_GENERATED_ASM_SRC
+    crypto/aes/asm/aesni-mb-x86_64.s
+    crypto/aes/asm/aesni-sha1-x86_64.s
+    crypto/aes/asm/aesni-sha256-x86_64.s
+    crypto/bn/asm/rsaz-avx2.s
+    crypto/bn/asm/rsaz-x86_64.s
+    crypto/bn/asm/x86_64-gf2m.s
+    crypto/bn/asm/x86_64-mont.s
+    crypto/bn/asm/x86_64-mont5.s
+    crypto/ec/asm/ecp_nistz256-x86_64.s
+    crypto/ec/asm/x25519-x86_64.s
+    crypto/md5/asm/md5-x86_64.s
+    crypto/modes/asm/aesni-gcm-x86_64.s
+    crypto/modes/asm/ghash-x86_64.s
+    crypto/sha/asm/keccak1600-x86_64.s
+    crypto/sha/asm/sha1-mb-x86_64.s
+    crypto/sha/asm/sha1-x86_64.s
+    crypto/sha/asm/sha256-mb-x86_64.s
+    crypto/sha/asm/sha256-x86_64.s
+    crypto/sha/asm/sha512-x86_64.s)
+
+set(OPENSSL_3_GENERATED_ASM_SRC
+    crypto/aes/asm/aes-x86_64.s crypto/aes/asm/bsaes-x86_64.s
+    crypto/bn/asm/rsaz-avx512.s)
+
 if (OE_SGX)
-  set(GENERATED_ASM_SRC
-      ${OPENSSL_BUILD}/crypto/aes/aesni-mb-x86_64.s
-      ${OPENSSL_BUILD}/crypto/aes/aesni-sha1-x86_64.s
-      ${OPENSSL_BUILD}/crypto/aes/aesni-sha256-x86_64.s
-      ${OPENSSL_BUILD}/crypto/bn/rsaz-avx2.s
-      ${OPENSSL_BUILD}/crypto/bn/rsaz-x86_64.s
-      ${OPENSSL_BUILD}/crypto/bn/x86_64-gf2m.s
-      ${OPENSSL_BUILD}/crypto/bn/x86_64-mont.s
-      ${OPENSSL_BUILD}/crypto/bn/x86_64-mont5.s
-      ${OPENSSL_BUILD}/crypto/ec/ecp_nistz256-x86_64.s
-      ${OPENSSL_BUILD}/crypto/ec/x25519-x86_64.s
-      ${OPENSSL_BUILD}/crypto/md5/md5-x86_64.s
-      ${OPENSSL_BUILD}/crypto/modes/aesni-gcm-x86_64.s
-      ${OPENSSL_BUILD}/crypto/modes/ghash-x86_64.s
-      ${OPENSSL_BUILD}/crypto/sha/keccak1600-x86_64.s
-      ${OPENSSL_BUILD}/crypto/sha/sha1-mb-x86_64.s
-      ${OPENSSL_BUILD}/crypto/sha/sha1-x86_64.s
-      ${OPENSSL_BUILD}/crypto/sha/sha256-mb-x86_64.s
-      ${OPENSSL_BUILD}/crypto/sha/sha256-x86_64.s
-      ${OPENSSL_BUILD}/crypto/sha/sha512-x86_64.s)
-
-  set(PATCHED_ASM_SRC
-      ${OPENSSL_BUILD}/crypto/aes/aesni-x86_64.s
-      ${OPENSSL_BUILD}/crypto/aes/vpaes-x86_64.s
-      # Needed by the OPENSSL_CPUID_OBJ compilation definition.
-      ${OPENSSL_BUILD}/crypto/x86_64cpuid.s)
+  foreach (source_file ${OPENSSL_GENERATED_ASM_SRC})
```

Comment:
Can you check if we can use
list(TRANSFORM <list> <APPEND|PREPEND> <value> ...) (if our cmake version supports it on both Windows and Linux)?

https://cmake.org/cmake/help/latest/command/list.html